Google Books.js

```diff
@@ -58,62 +61,74 @@ https://play.google.com/store/books/details/Adam_Smith_The_Wealth_of_Nations?id=
 
-var singleRe = /^https?:\/\/(?:books|www|play)\.google\.[a-z]+(?:\.[a-z]+)?(?:\/store)?\/books(?:\/.*)?\?(?:[^q].*&)?(id|vid)=([^&]+)/i;
 
 function detectWeb(doc, url) {
-	if (singleRe.test(url)) {
+	if (url.search(/[&?]v?id=/) != -1) {
 		return "book";
 	}
-	else {
+	else if (url.includes("/books/edition/")) {
+		return "book";
+	}
+	else if (getSearchResults(doc, true)) {
 		return "multiple";
 	}
+	return false;
 }
 
-var itemUrlBase;
-function doWeb(doc, url) {
-	// get local domain suffix
-	var psRe = new RegExp("https?://(books|www|play).google.([^/]+)/");
-	var psMatch = psRe.exec(url);
-	var suffix = psMatch[2];
-	itemUrlBase = "/books?id=";
-	
-	var m = singleRe.exec(url);
-	if (m && m[1] == "id") {
-		ZU.doGet("//books.google.com/books/feeds/volumes/" + m[2], parseXML);
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	// TODO: adjust the CSS selector
+	var rows = doc.querySelectorAll('div.bHexk>a, div.Q9MA7b>a');
+	for (let row of rows) {
+		let href = row.href;
+		// TODO: check and maybe adjust
+		let title = ZU.trimInternal(text(row, 'h3, div'));
+
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
 	}
-	else if (m && m[1] == "vid") {
-		var itemLinkWithID = ZU.xpath(doc, '/html/head/link[@rel="canonical"]')[0].href;
-		m = singleRe.exec(itemLinkWithID);
-		ZU.doGet("//books.google.com/books/feeds/volumes/" + m[2], parseXML);
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
 	}
 	else {
-		var items = getItemArrayGB(doc, doc, 'google\\.' + suffix + '/books\\?id=([^&]+)', '^(?:All matching pages|About this Book|Table of Contents|Index)');
-		// Drop " - Page" thing
-		for (let i in items) {
-			items[i] = items[i].replace(/- Page [0-9]+\s*$/, "");
-		}
-		Zotero.selectItems(items, function (items) {
-			if (!items) Z.done();
-			var baseurl = url.match(psRe)[0];
-			var newUris = [];
-			for (let i in items) {
-				// the singleRe has the full URL - we may only be getting the URL w/o host correct for that.
-				if (i.search(psRe) === -1) {
-					i = baseurl.replace(/\/$/, "") + i;
-				}
-				var m = singleRe.exec(i);
-				newUris.push("//books.google.com/books/feeds/volumes/" + m[2]);
-			}
-			ZU.doGet(newUris, parseXML);
-		});
+		scrape(doc, url);
 	}
 }
-	
+
+function scrape(doc, url) {
+	var id;
+	// New books format:
+	if (url.includes("/books/edition/")) {
+		id = url.match(/\/books\/edition.*\/([^/]+)(?:\?.+)?$/)[1];
```

Comment:
The regex captures from the last slash to ? or the end of the URL -- both are possible. Not sure I see the difference?

Reply:
The regex itself does IMO not say to take the last slash, because more slashes could be captured afterwards in the `.+` operator, but one could also make this more explicit by using `[^/]+` instead. (Or does greediness of the operator makes this unique anyway? But not sure we should rely on that...)

Reply:
Yeah, this relies on the greediness of `.*` but your version is simpler so going with that